v3/src/execution/index.ts

```diff
@@ -2,10 +2,164 @@ export {
   processGraphQLRequest,
   GraphQLRequestContext,
   GraphQLRequestPipelineConfig,
-} from "apollo-server-core/dist/requestPipeline";
-export { GraphQLRequest } from 'apollo-server-types';
+} from 'apollo-server-core/dist/requestPipeline';
+import { GraphQLRequest, VariableValues } from 'apollo-server-types';
+import {
+  DocumentNode,
+  parse,
+  GraphQLError,
+  GraphQLSchema,
+  validate,
+  execute,
+  ExecutionResult,
+  separateOperations,
+} from 'graphql';
+export { GraphQLRequest };
 
 // TODO(AS3) I'm not sure if this is execution.  Perhaps, a top-level export.
 export { GraphQLSchemaModule } from 'apollo-graphql';
 
 export { Context, ContextFunction } from 'apollo-server-core';
+
+/** Options for {@link processGraphQLRequest} */
+interface ProcessRequestInput {
+  request: GraphQLRequest;
+  schema: GraphQLSchema;
+}
+
+/**
+ * Process a GraphQLRequest. This includes parsing, validation,
+ * and execution of a query against a provided schema.
+ *
+ * @param args
+ * @param args.request - A GraphQLRequest object consisting of a query string and, optionally, an operationName, and variables
+ * @param args.schema - A GraphQLSchema to validate and execute the request against
+ *
+ * @returns A Promise consisting of data and errors from execution
+ * @throws GraphQLError on parse failure
+ * @throws GraphQLError[] on validation failure
```

Comment:
I'm uncertain about a couple things here, but I have some ideas.

1. Do we want this function to _throw_ anything? I think I like this, but it could get confusing since it can both `return { errors: GraphQLError[] }` or `throw GraphQLError | GraphQLError[]`. I chose this because we get a semantic distinction between throwing errors and returning them, but I'm unsure if it's important for us to be aware of or maintain that distinction. The difference being: a thrown error means we didn't execute at all, whereas a _returned_ array of errors means there were errors during execution.

2. If we appreciate this distinction, does it make sense to throw different types as this currently does? We could certainly just always throw `GraphQLError[]` instead by stuffing a parse error into an array first. Nothing really went into the decision to throw one vs. an array, besides that that's what `graphql`'s functions give us when we `parse` vs. `validate`.

Reply:
Alternatively, another means for maintaining the semantic difference would be to return these various errors in their own keys. I'm not too into it, but I can't deny that it's practical.

```js
return { parseError, validationErrors, executionErrors, data }
```

Reply:
Really we'd have a big union of a return type due to early returns, which I actually kinda like!
```ts
type ReturnType<TData> =
  | { parseError: GraphQLError }
  | { validationErrors: GraphQLError[] }
  | { data: TData, errors: GraphQLError[] };
```

Reply:
Returning all errors in an `{ errors: GraphQLError[] }` object now.